clients/js/src/compat.ts

```diff
@@ -0,0 +1,289 @@
+          if (prop === 'sendTransaction') {
+            return (async (signedTx) => {
+              const [ptTx, chainId] = await Promise.all([
+                signedTx,
+                signer.getChainId(),
+              ]);
+              const [encTx] = await encrypt(ptTx, chainId, opts);
+              return provider.sendTransaction(encTx);
+            }) as EthersProvider['sendTransaction'];
+          }
+          if (prop === 'call') {
+            return hookEthersCall(
+              signer,
+              provider.call.bind(provider) as EthersCall<string>,
+            );
+          }
+          if (prop === 'estimateGas') {
+            return hookEthersCall(
+              signer,
+              provider.estimateGas.bind(provider) as EthersCall<BigNumber>,
+            );
+          }
+          return Reflect.get(target, prop).bind(target);
+        },
+      }),
+    ) as T;
+  }
+
+  return new Proxy(upstream, {
+    get(target, prop) {
+      if (prop !== 'request') return Reflect.get(target, prop).bind(target);
+      const signer = new Web3Provider(upstream).getSigner();
+      return async (args: Web3ReqArgs) => {
+        const [params, meta] = await prepareRequest(args, signer, opts);
+        // TODO: decryt call response
+        const res = await signer.provider.send(args.method, params);
+        if (
+          !args.method.endsWith('call') ||
+          !meta ||
+          typeof res.data !== 'string'
+        )
+          return res;
+        return decrypt(arrayify(res.data), meta);
+      };
+    },
+  });
+}
+
+async function prepareRequest(
+  { method, params }: Web3ReqArgs,
+  signer: JsonRpcSigner,
+  opts: ProviderOptions | undefined,
+): Promise<[unknown[], MetaEncryptedX25519DeoxysII?]> {
+  if (!Array.isArray(params)) return [[params]];
+
+  const enc = async (data: BytesLike, chainId?: number) =>
+    encrypt(data, chainId ?? (await signer.getChainId()), opts);
+
+  if (method.endsWith('sendRawTransaction')) {
+    return [[enc(params[0]), ...params.slice(1)]];
+  }
+
+  if (
+    (method.endsWith('call') || method.endsWith('estimateGas')) &&
+    (await callNeedsSigning(params[0]))
+  ) {
+    const chainId = await signer.getChainId();
+    const [encData, meta] = await enc(params[0].data ?? [], chainId);
+    params[0].data = encData;
+    return [
+      [prepareSignedCall(params[0], signer, { chainId }), ...params.slice(1)],
+      meta,
+    ];
+  }
+
+  if (
+    method.endsWith('sendTransaction') ||
+    method.endsWith('signTransaction')
+  ) {
+    if (isBytesLike(params[0].data)) {
+      params[0].data = (await enc(params[0]))[0];
+    }
+    return [params];
+  }
+
+  return [params];
+}
+
+async function encrypt(
+  data: BytesLike,
+  chainId: number,
+  opts: ProviderOptions | undefined,
+): Promise<[string, MetaEncryptedX25519DeoxysII?]> {
+  if (data.length === 0) return [hexlify(data)];
+
+  const [packed, meta] = await callformat.encodeCall(
+    {
+      method: '',
```

Comment:
```suggestion
      format: CALLFORMAT_ENCRYPTED_X25519DEOXYSII,
      method: '',
```

Reply:
this one doesn't need to be set. the inner one is plain, which is the default